src/lints/struct_now_doc_hidden.ron

```diff
@@ -0,0 +1,50 @@
+SemverQuery(
+    id: "struct_now_doc_hidden",
+    human_readable_name: "pub struct is now #[doc(hidden)]",
+    description: "A pub struct is now hidden from documentation effectively removing it from the crate's public api.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#hidden"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        struct_type @output @tag
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Struct {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        struct_name: name @output
+                        struct_type @filter(op: "=", value: ["%struct_type"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                            public_api @filter(op: "!=", value: ["$true"])
+                        }
+
+                        span_: span @optional {
+                            filename @output
+                            begin_line @output
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+    },
+    error_message: "A publicly visible struct is now hidden via #[doc(hidden)].",
```

Comment:
Small tweak suggestion:
```suggestion
    error_message: "A pub struct is now #[doc(hidden)], removing it from the crate's public API.",
```
Many users might not know why `doc(hidden)` is relevant and being checked here, so we want to mention that the crate's public API has changed due to `doc(hidden)`.

Reply:
Good point, when is description shown then?

Reply:
Currently, it isn't shown in the CLI, but the plan is to eventually create a feature analogous to `rustc --explain <error-code>` that shows the description.

In the meantime, we link to the lint file's URL when lints are raised and users can look up the description that way.